### PR TITLE
Loosen some lifetime dependence diagnostics

### PR DESCRIPTION
**Pull request by meg-gupta**
This change removes diagnostic for lifetime dependence on ~Escapable & _BitwiseCopyable type.

Also includes changes to _BitwiseCopyable to be ~Escapable when the NonEscapableTypes feature is enabled.

And changes that add build-script option and lit config for building the compiler and testing with NonEscapableTypes feature.

rdar://123870480

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci smoke test

**Comment by meg-gupta** [comment]
@swift-ci smoke test

File: lib/Sema/LifetimeDependence.cpp [hunk 1]
```diff
@@ -159,12 +159,19 @@ LifetimeDependenceInfo::fromTypeRepr(AbstractFunctionDecl *afd, Type resultType,
...
     }
 
-    if (ctx.LangOpts.hasFeature(Feature::BitwiseCopyable)) {
-      auto *bitwiseCopyableProtocol =
-          ctx.getProtocol(KnownProtocolKind::BitwiseCopyable);
-      if (bitwiseCopyableProtocol && mod->checkConformance(type, bitwiseCopyableProtocol)) {
-        diags.diagnose(loc, diag::lifetime_dependence_on_bitwise_copyable);
-        return true;
+    // Diagnose when we have lifetime dependence on a type that is
+    // BitwiseCopyable & Escapable ~Escapable types are non-trivial in SIL and
```
Comment (atrick) [inline lib/Sema/LifetimeDependence.cpp:163]: > BitwiseCopyable & Escapable ~Escapable types

Missing punctuation

File: test/Sema/explicit_lifetime_dependence_specifiers.swift [hunk 2]
```diff
@@ -6,6 +6,14 @@ struct Container {
   let ptr: UnsafeRawBufferPointer
 }
 
+struct AnotherBufferView : ~Escapable {
```
Comment (atrick) [inline test/Sema/explicit_lifetime_dependence_specifiers.swift:9]: I AnotherBufferView inferred to be BitwiseCopyable? Do you want to specify BitwiseCopyable here to be explicit?

**Review by atrick — APPROVED** [review]
LGTM. 2 comments...

File: lib/Sema/LifetimeDependence.cpp (hunk 1 above)
Comment (meg-gupta) [reply lib/Sema/LifetimeDependence.cpp:182]: Done.

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test

**Review by nate-chandler — APPROVED** [review]
Thank you!

**Comment by meg-gupta** [comment]
@swift-ci smoke test

**Comment by meg-gupta** [comment]
@swift-ci smoke test

**Pull request merged**